### PR TITLE
Added support rabbitmq publishing.

### PR DESCRIPTION
**Pull request by mzabolotko**

### CODE REVIEW

File: Sources/Kontur.Rabbitmq/AmqpMessageBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿
```
Comment (alexzaitzev) [inline Sources/Kontur.Rabbitmq/AmqpMessageBuilder.cs]: Looks like extra new line

File: Sources/Kontur.Rabbitmq/AmqpSubscriptionBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            IAmqpSerializerFactory serializerFactory = new AmqpSerializerFactory("plain/text", defaultSerializer);
+            AmqpMessageBuilder amqpMessageBuilder = new AmqpMessageBuilder(propertyBuilder, router, serializerFactory);
+
+            TransformBlock<IMessage, AmqpMessage> amqpBuilderBlock =
+                new TransformBlock<IMessage, AmqpMessage>(
+                    (Func<IMessage, AmqpMessage>)amqpMessageBuilder.Build);
+
+            var factory = new ConnectionFactory();
+            factory.Uri = new Uri("amqp://localhost");
```
Comment (alexzaitzev) [inline Sources/Kontur.Rabbitmq/AmqpSubscriptionBuilder.cs:30]: It was intended to use hardcoded value of `localhost`?

File: Sources/Kontur.Rabbitmq/AmqpProperties.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        public string ReplyTo { get; set; }
+        public bool Persistent { get; set; }
+        public string MessageId { get; set; }
+        public IDictionary<string, string> Headers { get; set; }
+        public string CorrelationId { get; set; }
+        public string ContentType { get; set; }
+        public string ContentEncoding { get; set; }
+
+        public void CopyTo(IBasicProperties basicProperties)
```
Comment (alexzaitzev) [inline Sources/Kontur.Rabbitmq/AmqpProperties.cs:16]: For what purposes do we need this function in AmqpProperties?

File: Tests/Kontur.Rabbitmq.IntegrationTests/PublishingToAmqpFixture.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+            using (var subsciption = sut.WithRabbitMq<string>(cfg => 
+            {
+                cfg.RouteTo<string>("test", string.Empty);
+
+                return cfg;
+            }))
+            {
+                sut.EmitAsync("hello, world!", new Dictionary<string, string> { { "content-type", "text/plain" } }).Wait();
+                System.Threading.Thread.Sleep(50000);
```
Comment (alexzaitzev) [inline Tests/Kontur.Rabbitmq.IntegrationTests/PublishingToAmqpFixture.cs]: Sleep for 50 seconds? For what?

**Review by alexzaitzev — COMMENTED** [review]
Have some unclear moments in code

File: Sources/Kontur.Rabbitmq/AmqpProperties.cs (hunk 3 above)
Comment (mzabolotko) [reply Sources/Kontur.Rabbitmq/AmqpProperties.cs:16]: A common message is converted to a RabbitMQ message in the Dataflow AmqpBuilderBlock. Then this message will be sent by the Dataflow AmqpSenderBlock. The AmqpBuilderBlock and the AmqpSenderBlock forms Dataflow pipeline. 
The problem is to create RabbitMQ message with using BasicProperties the AmqpBuilderBlock need to know the model (AMQP channel). But the model is accessible only in the AmqpSenderBlock.

File: Sources/Kontur.Rabbitmq/AmqpSubscriptionBuilder.cs (hunk 2 above)
Comment (mzabolotko) [reply Sources/Kontur.Rabbitmq/AmqpSubscriptionBuilder.cs:30]: It example how to build pipes of Dataflow and integrate Kontour with Kontour.RabbitMQ. This method should be refactored aggressively. For example, the connection factory should be incapsulated into a ConnectionPool and a ConnectionPool shoud be shared among all AmqpSubscriptionBuilder instances. This method shoud provide a fluent interface to build subsriptions conveniently.

File: Tests/Kontur.Rabbitmq.IntegrationTests/PublishingToAmqpFixture.cs (hunk 4 above)
Comment (mzabolotko) [reply Tests/Kontur.Rabbitmq.IntegrationTests/PublishingToAmqpFixture.cs]: Fixed

File: Sources/Kontur.Rabbitmq/AmqpMessageBuilder.cs (hunk 1 above)
Comment (mzabolotko) [reply Sources/Kontur.Rabbitmq/AmqpMessageBuilder.cs]: Fixed

**Pull request merged**